packages/quantic/force-app/main/default/lwc/quanticResultsPerPage/e2e/pageObject.ts

```diff
@@ -0,0 +1,89 @@
+import type {Locator, Page, Request} from '@playwright/test';
+import {isUaCustomEvent} from '../../../../../../playwright/utils/requests';
+
+const resultsPerPageElementsSelectors = {
+  component: 'c-quantic-results-per-page',
+  componentError: 'c-quantic-component-error',
+  resultsPerPageTestId: 'results-per-page',
+  resultsPerPageOptionButton: 'c-quantic-number-button',
+  selectedResultsPerPageOptionButton: 'button.slds-button_brand',
+};
+
+export class ResultsPerPageObject {
+  constructor(public page: Page) {
+    this.page = page;
+  }
+
+  get component(): Locator {
+    return this.page.getByTestId(
+      resultsPerPageElementsSelectors.resultsPerPageTestId
+    );
+  }
+
+  get allResultsPerPageOptionsButtons(): Locator {
+    return this.component.locator(
+      resultsPerPageElementsSelectors.resultsPerPageOptionButton
+    );
+  }
+
+  get firstResultsPerPageOptionButton(): Locator {
+    return this.allResultsPerPageOptionsButtons.first();
+  }
```

Comment:
I think it's better if we keep our selector generic by accepting an index as parameter at it selects the option at the given index.

Reply:
Turns out I actually don't use it...